File: src/main/scala/li/cil/oc/integration/forestry/ModForestry.scala
package li.cil.oc.integration.forestry

import cpw.mods.fml.common.registry.GameRegistry
import li.cil.oc.Constants
import li.cil.oc.api.Driver
import li.cil.oc.common.recipe.Recipes
import li.cil.oc.integration.ModProxy
import li.cil.oc.integration.Mods

object ModForestry extends ModProxy {
  override def getMod = Mods.Forestry

  override def initialize() {
    Driver.add(new ConverterIAlleles)
    Driver.add(new ConverterIIndividual)
    Driver.add(ConverterItemStack)
    Driver.add(new DriverAnalyzer)
    Driver.add(new DriverBeeHouse)
    Driver.add(DriverUpgradeBeekeeper)
    Driver.add(DriverUpgradeBeekeeper.Provider)
    val multi = new li.cil.oc.common.item.Delegator()
    GameRegistry.registerItem(multi, "item.forestry")
    Recipes.addSubItem(new li.cil.oc.integration.forestry.item.UpgradeBeekeeper(multi), Constants.ItemName.BeekeeperUpgrade, "oc:beekeeperUpgrade")
  }
}

File: src/main/scala/li/cil/oc/integration/forestry/ModForestry.java
package li.cil.oc.integration.forestry;

import li.cil.oc.api.Driver;
import li.cil.oc.integration.IMod;
import li.cil.oc.util.mods.Mods;

public class ModForestry implements IMod {
    @Override
    public Mods.Mod getMod() {
        return Mods.Forestry();
    }

    @Override
    public void initialize() {
        Driver.add(new ConverterIAlleles());
        Driver.add(new ConverterIIndividual());
        Driver.add(new DriverBeeHouse());
    }
}
